### PR TITLE
Switch from multiple mysqls to one mariadb.

### PR DESCRIPTION
**Pull request by dwt**
This has several advantages:
- mariadb uses drastically less memory by default. On my machine each mysql instance uses at least one gig ram, while mariadb uses less than 40 megabytes.
- only one database again halves the resources used
- only one db is faster to start / stop
- only one db is easier to configure / reason about.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -129,20 +116,22 @@ services:
     volumes:
       - icingaweb:/data
 
-  icingaweb-mysql:
+  mysql:
+    # see https://hub.docker.com/_/mariadb
```
Comment (lippserd) [inline docker-compose.yml:120]: Please remove that comment.

File: docker-compose.yml [hunk 2]
```diff
@@ -129,20 +116,22 @@ services:
...
-      MYSQL_DATABASE: icingaweb
-      MYSQL_USER: icingaweb
-      MYSQL_PASSWORD: ${ICINGAWEB_MYSQL_PASSWORD:-icingaweb}
-    image: mysql
+      ICINGAWEB_MYSQL_PASSWORD: ${ICINGAWEB_MYSQL_PASSWORD}
+      ICINGADB_MYSQL_PASSWORD: ${ICINGADB_MYSQL_PASSWORD}
     logging: *default-logging
     volumes:
-      - icingaweb-mysql:/var/lib/mysql
+      # - ./mysql:/var/lib/mysql
```
Comment (lippserd) [inline docker-compose.yml:129]: Please remove that line.

File: docker-compose.yml [hunk 3]
```diff
@@ -129,20 +116,22 @@ services:
...
+      - mysql:/var/lib/mysql
+      - ./init-mysql.sh:/docker-entrypoint-initdb.d/init-mysql.sh
 
 volumes:
   icinga2:
-  icingadb-mysql:
+  mysql:
   icingaweb:
   icingaweb-mysql:
```
Comment (lippserd) [inline docker-compose.yml:137]: This volume could be removed as well, right? Also, I like things ordered 😆  So please put `mysql` last here.

File: init-mysql.sh [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    PASSWORD=$3
+    
+    mysql --user root --password=$MYSQL_ROOT_PASSWORD <<EOS
+CREATE DATABASE IF NOT EXISTS ${DB};
+CREATE USER IF NOT EXISTS  '${USER}'@'%' IDENTIFIED BY '${PASSWORD}';
+GRANT ALL ON ${DB}.* TO '${USER}'@'%';
+EOS
+}
+
+create_database_and_user icingaweb icingaweb ${ICINGAWEB_MYSQL_PASSWORD}
```
Comment (lippserd) [inline init-mysql.sh:15]: Looks like defaults for the env vars are missing, i.e. `icingaweb` and `icingadb`.

File: docker-compose.yml [hunk 5]
```diff
@@ -129,20 +116,22 @@ services:
...
-      MYSQL_PASSWORD: ${ICINGAWEB_MYSQL_PASSWORD:-icingaweb}
-    image: mysql
+      ICINGAWEB_MYSQL_PASSWORD: ${ICINGAWEB_MYSQL_PASSWORD}
+      ICINGADB_MYSQL_PASSWORD: ${ICINGADB_MYSQL_PASSWORD}
     logging: *default-logging
     volumes:
-      - icingaweb-mysql:/var/lib/mysql
+      # - ./mysql:/var/lib/mysql
+      - mysql:/var/lib/mysql
+      - ./init-mysql.sh:/docker-entrypoint-initdb.d/init-mysql.sh
```
Comment (lippserd) [inline docker-compose.yml:131]: Please move `init-mysql.sh` to `mysql` (rename it from `icingadb-mysql` and remove `privileges.sql`)  and mount `./env/mysql:/docker-entrypoint-initdb.d`.

**Review by lippserd — CHANGES_REQUESTED** [review]

**Comment by dwt** [comment]
Will do, I think I can get to that at the weekend or around Tuesday.

**Comment by lippserd** [comment]
> Will do, I think I can get to that at the weekend or around Tuesday.

 No rush!

File: docker-compose.yml (hunk 1 above)
Comment (dwt) [reply docker-compose.yml:120]: - [x] done

File: docker-compose.yml (hunk 2 above)
Comment (dwt) [reply docker-compose.yml:129]: - [x] done

File: docker-compose.yml (hunk 3 above)
Comment (dwt) [reply docker-compose.yml:137]: - [x] done

File: init-mysql.sh (hunk 4 above)
Comment (dwt) [reply init-mysql.sh:15]: I think these are better set in the docker-compose file, so I set it there.

File: docker-compose.yml (hunk 5 above)
Comment (dwt) [reply docker-compose.yml:131]: - [x] done

**Comment by dwt** [comment]
I've incorporated your feedback and rebased on master to make merging easier. It is a bit hard to test currently, because of [#icinga/docker-icingaweb2/78](https://github.com/Icinga/docker-icingaweb2/issues/78), but here's hope the this gets fixed eventually...

**Comment by dwt** [comment]
🎉

**Comment by lippserd** [comment]
By the way, thanks for the contribution! 👍

**Pull request merged**